corehq/apps/domain/decorators.py

```diff
@@ -231,6 +231,8 @@ def login_or_api_key_ex(allow_cc_users=False, allow_sessions=True):
 
 def _get_multi_auth_decorator(default, allow_token=False, allow_two_factor=True):
     def decorator(fn):
+        fn.allow_two_factor = allow_two_factor
```

Comment:
I like this approach.  Django also does some magic such that as long as you decorate the `dispatch` method, it'll copy this attr over to the generated view function.